### PR TITLE
SWEP: HUD help rework

### PR DESCRIPTION
**Pull request by lebroomer**
* Redesigned the HUD help rework
* Legacy support for the old function ```SWEP:AddHUDHelp()```: the help text is searched for "MOUSE1" and "MOUSE2" to add the respective icon
* It is now possible to add an arbitrary amount of lines to the help
* Added the possibility to visualize icons or keys together with the text

Fixes some of the problems mentioned in #575 

image

### CODE REVIEW

File: gamemodes/terrortown/entities/weapons/weapon_tttbase.lua [hunk 1]
```diff
@@ -239,42 +243,154 @@ if CLIENT then
...
+		local pad = 3
+		local pad2 = pad * 2
+
+		x = x - pad + 1
+		y = y - pad2 * 0.5 + 1
+
+		local key_box_w, key_box_h = draw.GetTextSize(key, "weapon_hud_help_key")
+
+		key_box_w = key_box_w + 3 * pad
+		key_box_h = key_box_h + pad2
```
Comment (TimGoll) [inline gamemodes/terrortown/entities/weapons/weapon_tttbase.lua:256]: missing empty line after this one

File: gamemodes/terrortown/entities/weapons/weapon_tttbase.lua [hunk 2]
```diff
@@ -239,42 +243,154 @@ if CLIENT then
...
+		secondaryfire = Key("+attack2", "MOUSE2"),
 		usekey = Key("+use", "USE")
 	}
 
+	---
+	-- Adds a help text for the weapon to the HUD.
+	-- TTT legacy function.
+	-- @param string primary_text first line of the help text
+	-- @param string secondary_text second line of the help text
+	-- @param bool translate should the text get translated
```
Comment (TimGoll) [inline gamemodes/terrortown/entities/weapons/weapon_tttbase.lua:327]: add a `[default=false]`

File: gamemodes/terrortown/entities/weapons/weapon_tttbase.lua [hunk 3]
```diff
@@ -239,42 +243,154 @@ if CLIENT then
...
+	-- @param table extra_params parameters for @{Lang.GetParamTranslation}
```
Comment (TimGoll) [inline gamemodes/terrortown/entities/weapons/weapon_tttbase.lua:328]: add an `[opt]`

File: gamemodes/terrortown/entities/weapons/weapon_tttbase.lua [hunk 4]
```diff
@@ -239,42 +243,156 @@ if CLIENT then
...
+	end
+
+	---
+	-- Adds an additional line to the help text.
+	-- @{SWEP:AddTTT2HUDHelp} needs to be called first
+	-- @param string text text to be displayed on the line
+	-- @param Material|string|nil icon_or_key icon or description for the concerning key
+	-- @realm client
+	function SWEP:AddHUDHelpLine(text, icon_or_key)
+		if not self.HUDHelp then
```
Comment (Alf21) [inline gamemodes/terrortown/entities/weapons/weapon_tttbase.lua:384]: one line please :)

File: gamemodes/terrortown/entities/weapons/weapon_tttbase.lua [hunk 5]
```diff
@@ -239,42 +243,156 @@ if CLIENT then
...
+			return
+		end
+
+		--replace MOUSE1/MOUSE2 strings with respective icons
+		icon_or_key = isstring(icon_or_key) and icon_or_key == "MOUSE1" and icon_help_primary or icon_or_key
```
Comment (Alf21) [inline gamemodes/terrortown/entities/weapons/weapon_tttbase.lua:389]: regarding double `isstring` check and double structure, I'd recommend using a simple if-elseif structure here. So we can remove the `or` logic as well.

Comment (Alf21) [reply gamemodes/terrortown/entities/weapons/weapon_tttbase.lua:389]:
```suggestion
		if isstring(icon_or_key) then
				if icon_or_key == "MOUSE1" then
						icon_or_key = icon_help_primary
				elseif icon_or_key == "MOUSE2" then
						icon_or_key = icon_help_secondary
				end
		end
```

File: gamemodes/terrortown/entities/weapons/weapon_tttbase.lua [hunk 6]
```diff
@@ -239,42 +243,160 @@ if CLIENT then
...
+
+		--replace MOUSE1/MOUSE2 strings with respective icons
+		if isstring(icon_or_key) then
+			if icon_or_key == "MOUSE1" then
+				icon_or_key = icon_help_primary
+			elseif icon_or_key == "MOUSE2" then
+				icon_or_key = icon_help_secondary
+			end
+		end
+		icon_or_key = isstring(icon_or_key) and icon_or_key == "MOUSE2" and icon_help_secondary or icon_or_key
```
Comment (Alf21) [inline gamemodes/terrortown/entities/weapons/weapon_tttbase.lua:394]: Yea the suggestions didn't supported multiline, so this line have to be removed for sure :D

**Pull request merged**